upgrades kibana & es version

tsop14 opened the pull request:
This was motivated by wanting to use the ✨improved✨ dashboard features in newer versions of Kibana, [4.5](https://www.elastic.co/blog/kibana-4-5-0-released), e.g. Assign labels to metric aggregations

This requires ES 2.3 so this is being upgraded too.

tsop14 committed:
updates kibana, ES and plugins

tsop14 committed:
updates elastic installation as per docs for 2.0

cb372 commented:
An arguably more important reason to upgrade ES: S3 backup is currently broken on Java 8u60+, due to https://github.com/elastic/elasticsearch-cloud-aws/issues/233. I have confirmed that the calls to the S3 API fail with `AmazonS3Exception: AWS authentication requires a valid Date or x-amz-date header`.

The ES upgrade will result in a corresponding upgrade of the AWS plugin, which should fix this issue.

Commit:
Remove obsolete setting from elasticsearch.yml

This was causing ES to fail to start up.

The setting is no longer needed anyway. ES 2.x has sensible defaults
regarding script execution. The default settings are:

```
script.inline: sandbox
script.indexed: sandbox
script.file: true
```

Reference:
https://www.elastic.co/guide/en/elasticsearch/reference/2.3/modules-scripting.html

Commit:
Give Kibana ownership of its own directory

It was failing to start up with the following error:

```
permission denied, open '/opt/kibana/optimize/.babelcache.json'
```

Commit:
Set ES advertised IP to the EC2 private IP address

It's currently advertising 127.0.0.1, so the nodes can't discover each
other
